### PR TITLE
Added drawer

### PR DESCRIPTION
**Pull request by rithulr808**

### CODE REVIEW

File: src/components/Drawer/drawer.module.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
+@keyframes translate {
+    from { transform: translate(var(--drawer-position-from-x) , var(--drawer-position-from-y)); overflow: auto; }
+    to { transform: translate(var(--drawer-position-to-x) , var(--drawer-position-to-y)); overflow: auto; }
+}
+
+@keyframes translate-reverse {
+    from { transform: translate(var(--drawer-position-from-x) , var(--drawer-position-from-y)); overflow: auto; }
+    to { transform: translate(var(--drawer-position-to-x) , var(--drawer-position-to-y)); overflow: auto; }
+}
```
Comment (Advaith3600) [inline src/components/Drawer/drawer.module.scss:9]: redundant keyframe definition

File: src/components/Drawer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  return  shouldRenderChild ? (
+      <DocumentPortal>
+          <div className="dsr-relative dsr-z-7000">
+              <section
+                  className={clsx([
+                    'dsr-fixed dsr-top-0 dsr-left-0 dsr-w-screen dsr-h-screen dsr-flex',
+                    'dsr-backdrop-filter dsr-backdrop-blur-sm dsr-bg-black dsr-bg-opacity-30',
+                    getPositionAlignmentParent,
+                  ],
+                  )}
```
Comment (Advaith3600) [inline src/components/Drawer/index.tsx:93]: can be in the same line as of `]`

File: src/components/Drawer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+              >
+                  <div
+                      className={clsx([
+                        'dsr-relative dsr-shadow-lg dsr-sm:w-auto dsr-w-full',
+                        getPositionAlignmentChild,
+                        getPositionAnimation,
+                      ])}
+                      style={{
+                        background: theme?.background,
+                        color: theme?.color,
```
Comment (Advaith3600) [inline src/components/Drawer/index.tsx:104]: remove these from styles and use tailwind classes

File: src/index.ts [hunk 4]
```diff
@@ -24,6 +24,7 @@ export { default as AccordionGroup } from './components/AccordionGroup';
 export { default as InfiniteLoader } from './components/InfiniteLoader';
 export { default as CircularProgress } from './components/CircularProgress';
 export { default as ConfirmationDialog } from './components/ConfirmationDialog';
+export { default as Drawer } from './components/Drawer';
```
Comment (Advaith3600) [inline src/index.ts:27]: order it by character length

**Review by Advaith3600 — CHANGES_REQUESTED** [review]

File: src/components/Drawer/drawer.module.scss (hunk 1 above)
Comment (rithulr808) [reply src/components/Drawer/drawer.module.scss:9]: it stops working when translate-reverse is removed

**Review by Advaith3600 — APPROVED** [review]

**Pull request merged**